include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    /* TimelineSemaphorePropertiesKHR *//* VK_KHR_timeline_semaphore *//* MOVED TO Vulkan 1.2 Core  */
+    /* VK_KHX_multiview *//* replaced by VK_KHR_multiview */
+
+    /* MultiviewPerViewAttributesPropertiesNVX *//* VK_NVX_multiview_per_view_attributes */
+    // VkBool32           perViewPositionAllComponents;
+
+    /* VK_NVX_raytracing *//* Preview Extension of raytracing, useless*/
+    
+    /* CooperativeMatrixPropertiesNV *//* VK_NV_cooperative_matrix */
+    // VkShaderStageFlags    cooperativeMatrixSupportedStages;
+
+    /* DeviceGeneratedCommandsPropertiesNV *//* VK_NV_device_generated_commands */
+    //uint32_t           maxGraphicsShaderGroupCount;
+    //uint32_t           maxIndirectSequenceCount;
+    //uint32_t           maxIndirectCommandsTokenCount;
+    //uint32_t           maxIndirectCommandsStreamCount;
+    //uint32_t           maxIndirectCommandsTokenOffset;
+    //uint32_t           maxIndirectCommandsStreamStride;
+    //uint32_t           minSequencesCountBufferOffsetAlignment;
+    //uint32_t           minSequencesIndexBufferOffsetAlignment;
+    //uint32_t           minIndirectCommandsBufferOffsetAlignment;
+
+    /* FragmentShadingRateEnumsPropertiesNV *//* VK_NV_fragment_shading_rate_enums */
+    // VkSampleCountFlagBits    maxFragmentShadingRateInvocationCount;
+
+    /* MeshShaderPropertiesNV *//* VK_NV_mesh_shader */
+    //uint32_t           maxDrawMeshTasksCount;
+    //uint32_t           maxTaskWorkGroupInvocations;
+    //uint32_t           maxTaskWorkGroupSize[3];
+    //uint32_t           maxTaskTotalMemorySize;
+    //uint32_t           maxTaskOutputCount;
+    //uint32_t           maxMeshWorkGroupInvocations;
+    //uint32_t           maxMeshWorkGroupSize[3];
+    //uint32_t           maxMeshTotalMemorySize;
+    //uint32_t           maxMeshOutputVertices;
+    //uint32_t           maxMeshOutputPrimitives;
+    //uint32_t           maxMeshMultiviewViewCount;
+    //uint32_t           meshOutputPerVertexGranularity;
+    //uint32_t           meshOutputPerPrimitiveGranularity;
+
+    /* VK_NV_ray_tracing *//* useless because of VK_KHR_ray_tracing_pipeline */
+
+    /* ShaderSMBuiltinsPropertiesNV *//* VK_NV_shader_sm_builtins */
+    //uint32_t           shaderSMCount;
+    //uint32_t           shaderWarpsPerSM;
+
+    /* ShadingRateImagePropertiesNV *//* VK_NV_shading_rate_image */
+    //VkExtent2D         shadingRateTexelSize;
+    //uint32_t           shadingRatePaletteSize;
+    //uint32_t           shadingRateMaxCoarseSamples;
+
+    /* FragmentDensityMapOffsetPropertiesQCOM *//* VK_QCOM_fragment_density_map_offset */
+    //VkExtent2D         fragmentDensityOffsetGranularity;
+
+    /* Nabla */
+    uint32_t maxBufferSize;
+    uint32_t maxOptimallyResidentWorkgroupInvocations = 0u; //  its 1D because multidimensional workgroups are an illusion
+    uint32_t maxResidentInvocations = 0u; //  These are maximum number of invocations you could expect to execute simultaneously on this device.
```

Comment:
You could actually provide this for

## Nvidia

https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceShaderSMBuiltinsPropertiesNV.html

Equation is simple
```cpp
VkPhysicalDeviceShaderSMBuiltinsPropertiesNV smBuiltinsPropertiesNV;

constexpr auto invocationsPerWarp = 32u; // unless Nvidia changed something recently
retval.maxResidentInvocations  = smBuiltinsPropertiesNV.shaderSMCount*smBuiltinsPropertiesNV.shaderWarpsPerSM*invocationsPerWrap;
```

## AMD

The equivalent to NVidia's
`VkPhysicalDeviceShaderCoreProperties2AMD::activeComputeUnitCount`

How many invocations fit on a CU (the warp per SM multiplied with invocations per warp) depends on the architecture though, no need to be precise, just find out whats the highest possible number.
_(It's not a huge efficiency problem to launch too many invocations for our compute tasks, the problem is too little or VASTLY too many.)_

https://www.khronos.org/registry/vulkan/specs/1.3-extensions/man/html/VkPhysicalDeviceShaderCoreProperties2AMD.html

## Default

Find out what the beefiest Intel GPU's EU count is and how many invocations it could possibly run simultaneously and set the default to that, the current default I have right now is that number but for an RTX 3090 Ti.